ra/ra.go

```diff
@@ -1423,19 +1396,57 @@ func (ra *RegistrationAuthorityImpl) NewOrder(ctx context.Context, req *rapb.New
 	if err != nil {
 		return nil, err
 	}
+	for _, name := range parsedCSR.DNSNames {
```

Comment:
Rather than operating off parsedCSR.DNSNames, we should downcase and uniquify them (`UniqueLowerNames`), and work off of that, or reject CSRs that contain duplicates (accounting for downcasing). Otherwise some of the assumptions below will be violated.